### PR TITLE
Replace scalaz with fs2

### PR DESCRIPTION
**Pull request by aeons**
This PR replaces scalaz stream with fs2.

Notes:
* I haven't updated the version in build.sbt
* I have removed 2.10 as a target version, since fs2 is 2.11+ only
* I kept `ByteVector` support, even though it looked like that was something inherited from scalaz
* I added an `Absorbable` instance for `NonEmptyChunk[Byte]`. This should probably be for a general `Chunk[Byte]`, but I couldn't figure that out.
* I have no idea what I'm doing, so the code could use a thorough review

Thanks to @mpilquist for helping me with the conversion.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,38 @@
...
 ```Scala
-import jawnstreamz._
+import jawnfs2._
 
 object Example extends App {
   // Pick your favorite supported AST (e.g., json4s, argonaut, etc.)
-  implicit val facade = jawn.ast.JawnFacade
-  // Read up to 64 bytes at a time
-  val chunkSizes: Process[Task, Int] = emitAll(Stream.continually(nextInt(64)))
+  implicit val facacde = jawn.ast.JawnFacade
```
Comment (rossabaker) [inline README.md]: `facade`

File: README.md [hunk 2]
```diff
@@ -1,45 +1,38 @@
...
 ```
 resolvers += "bintray/rossabaker" at "http://dl.bintray.com/rossabaker/maven"
 
-libraryDependencies += "org.http4s" %% "jawn-streamz" % "0.8.1"
+libraryDependencies += "org.http4s" %% "jawn-fs2" % "0.9.0"
 
 // Pick your AST: https://github.com/non/jawn#supporting-external-asts-with-jawn
-libraryDependencies += "org.jsawn" %% "jawn-ast" % "0.8.4"
+libraryDependencies += "org.jsawn" %% "jawn-ast" % "0.10.1"
 ```
```
Comment (rossabaker) [inline README.md]: I might leave the compatibility section as a historical note, and add `fs2-0.9 -> jawn-fs2-0.9`.

File: build.sbt [hunk 3]
```diff
@@ -51,20 +31,23 @@ developers := List(
...
+  "bintray/non" at "http://dl.bintray.com/non/maven"
 )
 
 scalacOptions ++= Seq(
   "-deprecation",
   "-feature"
 )
 
-val JawnVersion = "0.8.4"
+val JawnVersion = "0.10.1"
```
Comment (rossabaker) [inline build.sbt]: 0.10.2 was published today.

File: build.sbt [hunk 4]
```diff
@@ -51,20 +31,23 @@ developers := List(
...
+
+val Fs2Version = "0.9.1"
 
 libraryDependencies ++= Seq(
   "org.spire-math" %% "jawn-parser" % JawnVersion,
-  "org.spire-math" %% "jawn-ast" % JawnVersion % "test",
-  "org.scalaz.stream" %% "scalaz-stream" % scalazStreamVersion(scalazVersion.value),
-  "org.specs2" %% "specs2-core" % specs2Version(scalazVersion.value) % "test"
+  "org.scodec"     %% "scodec-bits" % "1.1.1",
```
Comment (rossabaker) [inline build.sbt]: I think this, and the `ByteVector` instance, can go away.  It's a great library, but `ByteVector` is not as necessary in an fs2 world with its improved chunking support.

**Comment by aeons** [comment]
Added some changes and figured out the Absorbable instance for Chunk.

Did you remove the comment on keeping the old versions for compatibility?

**Comment by rossabaker** [comment]
I did remove that comment, because it didn't seem to be attaching to the right place.  I'll go ahead and restore that section when I cut the release.  Thank you for this!

**Pull request merged**